added example for a blog that shows update dates

alexvoss opened the pull request:
these exist already in the template for the posts but were left out of the index pages for a tidier look.

I added the update date in parentheses to avoid questions about language or icon usage. I guess you won't rush into adding more strings that would require translations? It might be an idea to use the same icons as used in the post partial. LMK if you would prefer that.

alexvoss committed:
added example for a blog that shows update dates

these exist already in the template for the posts but were left out of
the index pages for a tidier look.

squidfunk commented:
Thanks for the example. The problem I see here is that the blog templates are all very experimental and will undergo fundamental changes in the near future, so this example will definitely break. We can merge and advertise it, but we will definitely need to revise it at some point. Could you also link it in the documentation using the shortcodes?

squidfunk commented on examples/blog-update/docs/README.md:
```diff
@@ -0,0 +1,22 @@
+# A blog with update dates shown
+
+This example shows how to add a small customization that will show
+then blob posts were [updated] on the blog index pages. (The blog page
```
Small typo: blob

squidfunk commented on examples/blog-update/overrides/partials/post.html:
```diff
@@ -0,0 +1,67 @@
+          <!-- start customization -->
+          {% if post.config.date.updated %}
+          <time datetime="{{ post.config.date.updated }}">
+            &nbsp;({{- post.config.date.updated | date -}})
```
That is actually a smell – `&nbsp` should not be necessary. If you remove the `-` in the placeholder you should be able to remove it, as those are used for whitespace control.

alexvoss replied:
I do need to study some Jinja... will do it just after my 100 pushups ;o)

squidfunk commented on examples/blog-update/overrides/partials/post.html:
```diff
@@ -0,0 +1,67 @@
+          </time>
+          <!-- start customization -->
+          {% if post.config.date.updated %}
+          <time datetime="{{ post.config.date.updated }}">
```
Please indent by 2 spaces. `if` conditions always start a new block.

squidfunk commented on examples/blog-update/overrides/partials/post.html:
```diff
@@ -0,0 +1,67 @@
+{#-
```
The banner should be removed, since this file was not automatically generated but manually overridden 😉

alexvoss replied:
I have used the space for a note about updates. Hope that makes sense. It is now mentioned in the README as well.

squidfunk commented on examples/blog-update/mkdocs.yml:
```diff
@@ -0,0 +1,40 @@
+  custom_dir: overrides
+
+watch:
+  - overrides
```
That is not necessary, I think. IMHO `mkdocs serve --watch-theme` is what you want.

alexvoss replied:
That is not how I read the documentation. That would not work if people did not use the name `overrides`. I assumed it meant the files in the theme, wherever it is resolved from but *not* overrides, useful that might be, though. In any case, would it not be easier to specify this in `mkdocs.yml` rather than having to remember doing it on the command-line? That is what I have always done.

squidfunk replied:
MkDocs will add the `custom_dir` directory to `config.theme.dirs`, [which are all watched when using `--watch-theme`](https://github.com/mkdocs/mkdocs/blob/828f4685f29dd9e986f18306d58d1cb383d00222/mkdocs/commands/serve.py#L94-L96). Thus adding the folder explicitly is unnecessary 😉

alexvoss replied:
That is good to know. Still leaves the question whether it is a good idea to add it to the watchlist in `mkdocs.yml` to ensure it is watched even when people forget to add the `--watch-theme` part. Is there any reason why doing the latter would be better? My way seems (naively) safer.

squidfunk replied:
Good point, we should leave it in `mkdocs.yml`.

alexvoss committed:
implemented PR feedback

- fixed typo
- removed &nbsp;
- replaced "don't edit" warning with message about updates
- added note on updates to README
- fixed indentation

squidfunk commented:
Okay so this looks good, but I have one big problem with this example: it uses a custom override of a blog partial. I thought about this, and I'm not sure we are ready to advertise this in our list of examples, because this specific partial might be changed drastically in the coming weeks, which might break many installations later. Thus, I'd opt for blog examples that do not change overrides for now, and we'll merge this PR later once I've had a change to revisit the templates. It's just that I want to keep things as stable as possible and that includes only advertising stuff that is considered stable (or putting an experimental flag behind it). Since partials cannot have flags, IMHO, we should postpone this specific example.

alexvoss commented:
Sure, no problem. I created the example in response to a discussion - are you ok it I point the OP to the branch with the warning that things will change?

squidfunk commented:
Yes, of course.

squidfunk commented:
@alexvoss could you add this as an example to our docs? For now, [here](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#writing-your-first-post) might be the best place, but we should definitely rewrite the documentation of the blog to be more use-case specific and hands-on.
